components/wifi/esp8266-driver/ESP8266/ESP8266.cpp

```diff
@@ -33,7 +33,9 @@
 
 #define TRACE_GROUP  "ESPA" // ESP8266 AT layer
 
+#ifndef ESP8266_DEFAULT_BAUD_RATE
```

Comment:
Modify https://github.com/ARMmbed/mbed-os/blob/master/components/wifi/esp8266-driver/mbed_lib.json
to have one extra config line like:
`"baudrate": {
            "help": "Baudrate for ESP8266, defaults to 115200",
            "value": 115200
        }
`
Then in code you can use "MBED_CONF_ESP8266_BAUDRATE" instead of defining "ESP8266_DEFAULT_BAUD_RATE"

Reply:
To avoid confusion perhaps this config could be called `uart_baudrate` or `serial_baudrate`, to avoid an impression that this directly influences the WiFi communication speed, which might be a primary interest of many users. 
I realize that the serial baudrate in fact throttles the module's communication speed, but I think we should be specific with the naming, whenever we can afford it :).